blog/2023-09-25-constructs-methods.md

```diff
@@ -0,0 +1,247 @@
+---
+title: Designing programmatic APIs for cloud infrastructure using constructs (part 1)
+description: Technical deep dive on some of the challenges of designing ergonomic APIs using the constructs framework
+authors: 
+  - rybickic
+tags: [winglang, cdk, constructs, api design, jsii, AWS CDK]
+hide_table_of_contents: true
+---
+
+One of the essential features of Winglang is being able to use familiar programming language tools like variables, for loops, and classes to describe the cloud resources that are part of your application.
+For example, cloud applications may rely on databases, queues, storage buckets, API endpoints, as well as compute platforms such as Kubernetes clusters or AWS Lambda.
+
+Today, Winglang uses **constructs**, a foundational framework that underpins several infrastructure-as-code libraries such as [AWS CDK], [CDKTF], and [cdk8s], to allow cloud components to be expressed and composed together into higher level abstractions.
```

Comment:
Maybe add something a bit more personal in first person to give a little background and context about yourself, about constructs (CDK) and about the Wing SDK.

Reply:
@eladb I don't feel like talking about myself in this kind of technical deep dive post feels right for my personal voice / writing style -- but I've taken the opportunity to tighten up the introduction and get to the point quicker. Let me know if this is okay...